lib/plek.rb

```diff
@@ -124,48 +116,37 @@ def website_root
     env_var_or_dev_fallback("GOVUK_WEBSITE_ROOT") { find("www") }
   end
 
-  # Find the base URL for assets.
-  #
-  # @return [URI::HTTPS,URI::HTTP,URI::Generic] The assets base URL.
-  def asset_uri
-    URI(asset_root)
-  end
-
-  # Find the base URL for the public website frontend.
-  #
-  # @return [URI::HTTPS,URI::HTTP,URI::Generic] The website base URL.
-  def website_uri
-    URI(website_root)
-  end
-
-  # TODO: clean up all references to these and then remove them.
   class << self
-    # This alias allows calls to be made in the old style:
-    #     Plek.current.find('foo')
-    # as well as the new style:
-    #     Plek.new.find('foo')
-    alias_method :current, :new
-
-    # Convenience wrapper.  The same as calling +Plek.new.find+.
-    # @see #find
-    def find(*args)
-      new.find(*args)
-    end
-
-    # Convenience wrapper.  The same as calling +Plek.new.find_uri+.
-    # @see #find_uri
-    def find_uri(*args)
-      new.find_uri(*args)
-    end
+    extend Forwardable
+
+    # @!method find
+    #   Convenience wrapper. The same as calling +Plek.new.find+.
+    #   @see #find
+    #   @return [String]
+    # @!method external_url_for
+    #   Convenience wrapper. The same as calling +Plek.new.external_url_for+.
+    #   @see #external_url_for
+    #   @return [String]
+    # @!method asset_root
+    #   Convenience wrapper. The same as calling +Plek.new.asset_root+.
+    #   @see #asset_root
+    #   @return [String]
+    # @!method website_root
+    #   Convenience wrapper. The same as calling +Plek.new.website_root+.
+    #   @see #website_root
+    #   @return [String]
+    def_delegators :new, :find, :external_url_for, :asset_root, :website_root
```

Comment:
Optional: it'd be awesome if we could avoid the penalty of instantiating and throwing away a new Plek on every call to these convenience methods. I don't know how hot a path it really is but it's certainly called a bunch when generating certain pages.

I don't know what the most idiomatic Ruby way to do that would be — maybe something along the lines of (pseudocode, probably not correct Ruby) :

```ruby
def_delegators :default, :find, :external_url_for, :asset_root, :website_root

def self.default
  @default_instance ||= self.new
end
```
?

Reply:
fwiw Plek is instantiated roughly 10k times in the Whitehall tests 😅

Reply:
That's a nice idea and it's often done in [Ruby via the Singleton module](https://www.rubyguides.com/2018/05/singleton-pattern-in-ruby/). I don't actually think it's likely to be of too great a benefit here though so I'm not too concerned -  my expectation is that the memory used for this is very modest and garbage collected straight away. 

I think it'd also add a few complications as Plek is used quite frequently in tests and any env var evaluations would no longer occur. Everything gets a bit more complex once you have global state held.

So I'd rather see it become a problem rather than try solve it optimistically. As the current pattern already instantiates plek each time this wouldn't increase allocations.

Reply:
Oh that's a good point about the tests. Yuck. Yeah agree it's not worth doing without measuring first then, and we shouldn't do it in this PR.